shared_model/interfaces/iroha_internal/transaction_sequence.hpp

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_TRANSACTION_SEQUENCE_HPP
+#define IROHA_TRANSACTION_SEQUENCE_HPP
+
+#include <boost/iterator/iterator_facade.hpp>
+#include <boost/range/any_range.hpp>
+#include <set>
+#include "interfaces/transaction.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    struct TransactionComparator {
+      bool operator()(const std::shared_ptr<Transaction> &lhs,
+                      const std::shared_ptr<Transaction> &rhs) const;
+    };
+
+    /**
+     * Transaction sequence is the collection of transactions, where
+     * transactions from the same batch are placed contiguously
+     */
+    using TransactionSequence =
```

Comment:
Using not guarantee that we store our collection in a proper way. Maybe rework it with a delegation? I mean, create a class with will hold collection with objects and getter for the collection.

Reply:
What do you mean by proper way? I think we should have iteratable interface with insert method which stores transactions in proper order. Set suits here. Any other requirements you see there?